pyomo/core/plugins/transform/deactivate_trivial_constraints.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: UTF-8 -*-
+"""Transformation to deactivate trivial constraints."""
+import textwrap
+from math import fabs
+
+from pyomo.core.base.constraint import Constraint
+from pyomo.core.kernel.component_set import ComponentSet
+from pyomo.core.kernel.numvalue import value
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+from pyomo.util.plugin import alias
+
+__author__ = "Qi Chen <https://github.com/qtothec>"
+
+
+class TrivialConstraintDeactivator(IsomorphicTransformation):
+    """Deactivates trivial constraints of form constant = constant."""
+
+    alias('core.deactivate_trivial_constraints',
+          doc=textwrap.fill(textwrap.dedent(__doc__.strip())))
+
+    def __init__(self):
+        """Initialize the transformation."""
+        super(TrivialConstraintDeactivator, self).__init__()
+        self.tolerance = 1E-13
+
+    def _apply_to(self, instance, tmp=False, ignore_infeasible=False):
```

Comment:
What does the "tmp" argument do?

Reply:
Specifies if the user has any intention of reversing the transformation in the future. I'm going to go clean up the documentation.